aioesphomeapi/_frame_helper/noise.py

```diff
@@ -168,7 +160,14 @@ def data_received(self, data: bytes) -> None:
                 return
 
             try:
-                self._dispatch(self, frame)
+                if self._state == NOISE_STATE_READY:
+                    self._handle_frame(frame)
+                elif self._state == NOISE_STATE_HELLO:
+                    self._handle_hello(frame)
+                elif self._state == NOISE_STATE_HANDSHAKE:
+                    self._handle_handshake(frame)
+                else:
+                    self._handle_closed(frame)
```

Comment:
I wrote this out so we it does the bytes check here to make sure we didn't have another problem deeper in the stack